pkg/azuredisk/controllerserver.go

```diff
@@ -524,7 +524,65 @@ func (d *Driver) GetCapacity(ctx context.Context, req *csi.GetCapacityRequest) (
 
 // ListVolumes return all available volumes
 func (d *Driver) ListVolumes(ctx context.Context, req *csi.ListVolumesRequest) (*csi.ListVolumesResponse, error) {
-	return nil, status.Error(codes.Unimplemented, "")
+	var err error
+	start := 0
+	if req.StartingToken != "" {
+		start, err = strconv.Atoi(req.StartingToken)
+		if err != nil {
+			return nil, status.Errorf(codes.Aborted, "ListVolumes starting token(%s) parsing with error: %v", req.StartingToken, err)
+		}
+		if start < 0 {
+			return nil, status.Errorf(codes.Aborted, "ListVolumes starting token(%d) can not be negative", start)
+		}
+	}
+
+	disks, derr := d.cloud.DisksClient.ListByResourceGroup(ctx, d.cloud.ResourceGroup)
+
+	if derr != nil {
+		return nil, status.Error(codes.Internal, fmt.Sprintf("Unknown list disks error: %v", derr.Error()))
+	}
+
+	if start != 0 && start >= len(disks) {
+		return nil, status.Errorf(codes.Aborted, "ListVolumes starting token(%d) is greater than total number of volumes", start)
+	}
+
+	maxEntries := len(disks) - start
+	if req.MaxEntries > 0 && int(req.MaxEntries) < maxEntries {
+		maxEntries = int(req.MaxEntries)
+	}
+	pageEnd := start + maxEntries
+
+	entries := []*csi.ListVolumesResponse_Entry{}
+
+	for i := start; i < pageEnd; i++ {
+		d := disks[i]
+		nodeList := []string{}
+
+		if d.ManagedBy != nil {
+			nodeList = append(nodeList, *d.ManagedBy)
```

Comment:
this is incorrect, `ManagedBy` string format is actually not nodeName for VMSS instance, let's remove this implementation in this PR(remove ), and only supports `LIST_VOLUMES` in this PR, and put implementation of `LIST_VOLUMES_PUBLISHED_NODES` in another PR.

Reply:
use transform like this, it will provide you the nodeName:
```
			attachedNode, err := vmset.GetNodeNameByProviderID(*disk.ManagedBy)
			if err != nil {
				return -1, err
			}
```

Reply:
Ok sure, thanks. Shall I still remove this part and take this in a separate PR?

Reply:
support `LIST_VOLUMES` and `LIST_VOLUMES_PUBLISHED_NODES` is ok, while current disk list could be incomplete since some disks could be in other resource group, do you know what's the impact if `ListVolumes` return incomplete disk list?

Reply:
AFAIK, ListVolumes is used by the sidecar csi-attacher to reconcile VolumeAttachment objects by comparing the state of VolumeAttachment objects with the ListVolumes response. if the list of disks is incomplete, reconciliation of the VolumeAttachment objects for the corresponding missing disks will be hampered.

Reply:
then disable the sanity test failure temporarily here:
https://github.com/kubernetes-sigs/azuredisk-csi-driver/blob/44dedf04edda6f5d22a625a0756a98044fbfb002/test/sanity/run-test.sh#L38

and then work on #622

also open an issue that `ListVolume` does not work if disks are created in external rg, I think finally we may use `d.cloud.KubeClient.CoreV1().PersistentVolumes()` to reimplement the ListVolumes logic, that could discover all disks.

Reply:
Issue to track the external resource group scenario : https://github.com/kubernetes-sigs/azuredisk-csi-driver/issues/632